src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -16,6 +17,14 @@
           Condition="'$(RestorePackages)'=='true' AND '$(IsTestProject)' == 'true' AND '$(TestRuntimeProjectJson)' != ''">
     <Exec Command="$(DnuRestoreCommand) &quot;$(TestRuntimeProjectJson)&quot;" StandardOutputImportance="Low" CustomErrorRegularExpression="^Unable to locate .*" />
   </Target>
+  
+  <Target Name="ValidateTestRuntimeDependencies"
+          AfterTargets="RestoreTestRuntimePackage"
```

Comment:
@ericstj @weshaggard I've been trying out a semaphore and I think it would be a good fix to make to eliminate some unnecessary restores but as far as I can tell it can't help with making validation only happen once. The main problem is because after writing a semaphore the task won't happen again unless the inputs change: if validation fails before touching the semaphore, the other projects don't know that the file's already failed validation and will also try to validate it. If the semaphore is touched before validating, it will report one error, but subsequent (root or project) builds will skip validation entirely and report success. While figuring this out I've added a semaphore for `RestoreTestRuntimePackage`, so I'll leave it in the PR, but I think I'll need to do the same thing as restore with root and project validation being distinct.